src/storage/src/mut_records.rs

```diff
@@ -145,34 +258,67 @@ impl FileRecords for MutFileRecords {
         self.base_offset
     }
 
-    fn get_file(&self) -> &File {
-        &self.f_sink.inner()
-    }
+    // the get_file method was never called, leaving it in makes
+    // the ownership management of the f_sink impossible to place in a mutex
+    // to manage access from send() writers and a async timed delay flush
+    // not sure how to resolve this
+
+    // fn get_file(&self) -> &File {
+    //
+    //     // &self.f_sink.inner()
+    //
+    //     block_on(async {
+    //         let am = self.f_sink.clone();
+    //         let f_sink = am.lock().await;
+    //         &f_sink.inner().clone()
+    //     })
+    // }
 
     fn get_path(&self) -> &Path {
         &self.path
     }
 
     fn as_file_slice(&self, start: Size) -> Result<AsyncFileSlice, IoError> {
-        self.f_sink
-            .slice_from(start as u64, self.f_sink.get_current_len() - start as u64)
+        block_on(async {
```

Comment:
avoid block_on

Reply:
The slice methods are synchronous in the FileRecords trait. I used block_on to limit the scope of changes, but if changing the trait is ok too then I can look at widening that.

Reply:
Yes, we should not be blocking on here since this is critical path